code/game/objects/items/tools/maintenance_tools.dm

```diff
@@ -246,7 +246,7 @@
 /obj/item/tool/weldingtool/afterattack(obj/O as obj, mob/user as mob, proximity)
 	if(!proximity)
 		return
-	if (istype(O, /obj/structure/reagent_dispensers/fueltank) && get_dist(src,O) <= 1)
+	if (O.reagents == "fuel")
```

Comment:
still needs to typecheck as a fueltank otherwise you will be able to fill it from beakers,  food, etc